widget/playground/src/utils/common.ts

```diff
@@ -17,3 +25,103 @@ export function shallowEqual<T>(
 
   return true;
 }
+
+export const tokenToString = (token: Asset) =>
+  `${token.symbol}-${token.blockchain}-${token.address ?? ''}`;
+
+export function removeDuplicates<T>(arr: T[], key?: keyof T): T[] {
+  if (!key) {
+    return Array.from(new Set(arr));
+  }
+
+  const seen = new Set<T[keyof T]>();
+  return arr.filter((item) => {
+    const identifier = item[key];
+    if (!seen.has(identifier)) {
+      seen.add(identifier);
+      return true;
+    }
+    return false;
+  });
+}
```

Comment:
This is an AI suggestion :smiley_cat:  what do you think about that? I think your code is more readable but this one is more simplified. :thinking: 
```ts
export function removeDuplicates<T>(arr: T[], key?: keyof T): T[] {
  const seen = new Set<T[keyof T]>();

  return arr.filter(item => !key || typeof item[key] !== 'undefined' && !seen.has(item[key]) && seen.add(item[key]));
}
```

Reply:
right.